src/algebra/algebra/spectrum.lean

```diff
@@ -245,6 +243,88 @@ begin
     exact ⟨unit_mem_mul_iff_mem_swap_mul.mp k_mem, k_neq⟩ },
 end
 
+open polynomial
+/-- This is half of the spectral mapping theorem for polynomials. We prove it separately
+because it holds over any field, whereas `spectrum.polynomial_spectrum` needs the field to
+be algebraically closed. -/
+theorem polynomial_subset (a : A) (p : polynomial 𝕜) :
+  (λ k, eval k p) '' (σ a) ⊆ σ (aeval a p) :=
+begin
+  rintros _ ⟨k, hk, rfl⟩,
+  let q := C (eval k p) - p,
+  have hroot : is_root q k, by simp only [eval_C, eval_sub, sub_self, is_root.def],
+  rw [←mul_div_eq_iff_is_root, ←neg_mul_neg, neg_sub] at hroot,
+  have aeval_q_eq : ↑ₐ(eval k p) - aeval a p = aeval a q,
+    by simp only [aeval_C, alg_hom.map_sub, sub_left_inj],
+  rw [mem_iff, aeval_q_eq, ←hroot, aeval_mul],
+  have hcomm := aeval_comm a (C k - X) (- (q / (X - C k))),
+  apply mt (λ h, (hcomm.is_unit_mul_iff.mp h).1),
+  simpa only [aeval_X, aeval_C, alg_hom.map_sub] using hk,
+end
+
+lemma aeval_list_prod_is_unit {R M : Type*} [comm_semiring R] [ring M] [algebra R M] {m : M}
+  {s : list (polynomial R)} (h : ∀ p ∈ s, is_unit (aeval m p)) :
+  is_unit (aeval m s.prod) :=
+begin
+  induction s,
+  { simp only [aeval_one, is_unit_one, list.prod_nil] },
+  { have u_hd := h s_hd (list.mem_cons_self s_hd s_tl),
+    have u_tl := s_ih (λ p p_mem, h p (list.mem_of_mem_tail p_mem)),
+    rw [list.prod_cons, aeval_mul, (aeval_comm m _ _).is_unit_mul_iff],
+    exact ⟨u_hd, u_tl⟩ }
+end
```

Comment:
I don't think this lemma should exist, since the proof is just:
```suggestion
lemma aeval_list_prod_is_unit {R M : Type*} [comm_semiring R] [ring M] [algebra R M] {m : M}
  {s : list (polynomial R)} (h : ∀ p ∈ s, is_unit (aeval m p)) :
  is_unit (aeval m s.prod) :=
begin
  rw alg_hom.map_list_prod,
  apply list.prod_is_unit,
  intros a m,
  obtain ⟨a, ha, rfl⟩ := list.mem_map.1 m,
  exact h _ ha,
end
```
If you really want to add a lemma here, it would be `list.prod_map_is_unit` which would handle the awkward `list.map` goal for you; but even that seems somewhat dubious.

Interesting, `list.prod_is_unit` is used in exactly one place, `exists_spectrum_of_is_alg_closed_of_finite_dimensional`. Is that lemma related to the `spectrum` in this file?

Reply:
fair enough, I can include it as a `have` in the proof of the main theorem.

Reply:
Regarding `exists_spectrum_of_is_alg_closed_of_finite_dimensional`: yes, I just saw that the other day. It is the same, but that theorem existed before the new spectrum. At some point we should either yank that theorem into this file (especially now that we are importing the `alg_closed` file), or at least prove the `spectrum` version of it here. Which do you think is more appropriate?